File: 프로그래머스/정렬/가장큰수.py
# 프로그래머스 가장 큰 수
# 21.02.01
from functools import cmp_to_key


def str_compare(a, b):
    if a+b > b+a: return 1
    elif a+b < b+a: return -1
    else: return 0


def solution(numbers):
    answer = ''
    strNum = [str(n) for n in numbers]
    strNum = sorted(strNum, key=cmp_to_key(str_compare), reverse=True)
    if strNum[0] == '0': answer = '0'
    else:
        for s in strNum:
            answer += s
    return answer

File: programmers/lv2/가장큰수.swift
// 프로그래머스 가장 큰 수
// 21.04.18

import Foundation

func solution(_ numbers:[Int]) -> String {
  var ans = ""
  
  let nums = numbers.sorted(by: {
    String($0) + String($1) > String($1) + String($0)
  })

  if nums[0] == 0 { ans = "0" }
  else {
    for i in nums { ans += String(i) }
  }
  
  return ans
}

print(solution([1, 10, 100]))
